src/util/i18nFormatting.js

```diff
@@ -0,0 +1,24 @@
+const formatFlags = {
+  EN_HANDLE_AN: 'en-handle-an',
+  CAPITALIZE: 'capitalize'
+}
+
+function getVariationOfAOrAn(value) {
+  const vowelishLetters = ['a', 'e', 'i', 'o', 'u', 'h'];
+  const isVowelish = vowelishLetters.find(l => value.substring(0, 1) === l);
+  return isVowelish ? 'an' : 'a';
+}
+
+// eslint-disable-next-line no-unused-vars
+export default function(value, format, lng) {
+  let newVal = value;
+  const inputFlags = format.split('|');
+  inputFlags.forEach((flag) => {
```

Comment:
Particularly combined with the `formatters` proposal below, a `reduce` would be very elegant here.